gamification/src/main/java/com/appcoins/wallet/gamification/repository/BdsPromotionsRepository.kt

```diff
@@ -30,7 +33,43 @@ class BdsPromotionsRepository(private val api: GamificationApi,
               .map { (promotions, walletOrigin) ->
                 mapErrorToUserStatsModel(promotions, walletOrigin, t)
               }
-              .onErrorReturn { mapErrorToUserStatsModel(t) }
+              .onErrorReturn { mapErrorToUserStatsModel(t, false) }
+        }
+  }
+
+  private fun getUserStats(wallet: String): Observable<UserStatusResponse> {
+    return Observable.concat(getUserStatsFromDB(wallet), getUserStatsFromAPI(wallet))
+  }
+
+  private fun getUserStatsFromDB(wallet: String): Observable<UserStatusResponse> {
+    return Single.zip(local.getPromotions(), local.retrieveWalletOrigin(wallet),
+        BiFunction { promotions: List<PromotionsResponse>, walletOrigin: WalletOrigin ->
+          Pair(promotions, walletOrigin)
+        })
+        .toObservable()
+        .map { (promotions, walletOrigin) ->
+          UserStatusResponse(promotions, walletOrigin, null, true)
```

Comment:
Not on you, but since we are doing a lot of changes here, we should create a model object instead of using Response object. This was being done wrongly before, but by adding the error and fromCache variables we need to create a model object. The "Response" object should contain only what the server returns, any additional logic, should be converted to a model object